rosbag2_test_common/include/rosbag2_test_common/subscription_manager.hpp

```diff
@@ -80,15 +80,23 @@ class SubscriptionManager
     return async(
       std::launch::async, [this]() {
         rclcpp::executors::SingleThreadedExecutor exec;
-        while (rclcpp::ok() && continue_spinning(expected_topics_with_size_)) {
+        auto start = std::chrono::high_resolution_clock::now();
+        while (rclcpp::ok() && continue_spinning(expected_topics_with_size_, start)) {
           exec.spin_node_some(subscriber_node_);
         }
       });
   }
 
 private:
-  bool continue_spinning(std::unordered_map<std::string, size_t> expected_topics_with_sizes)
+  bool continue_spinning(
+    std::unordered_map<std::string, size_t> expected_topics_with_sizes,
+    std::chrono::time_point<std::chrono::high_resolution_clock> start_time)
   {
+    auto current = std::chrono::high_resolution_clock::now();
+    if (current - start_time > std::chrono::seconds(10)) {
```

Comment:
What is "10 seconds" here?

Reply:
That's a 10 seconds time out. If the test messages are not arrived by then, the thread canceled. I needed that to avoid running in a Gtest timeout situation (60 seconds) at which point one does not know why the test timed out.